### PR TITLE
Yellow Ribbon Program support

### PR DESCRIPTION
**Pull request by jkassemi**
Provide the ability to upload Yellow Ribbon Program data provided by EDU
team. A new file upload option: "Yellow Ribbon Program Source" is
provided. General validation and conversion is supported, though some
fields that are only used on re-export of data source will need to be
worked through with the EDU team in more detail to ensure we have
consistent and correct input data.

Yellow Ribbon Program Source records are created for each row in the
input file. The Institution Builder is modified to create a new Yellow
Ribbon Program record for each program that can be associated with an
institution. Noticing that we're not seeing a full match to all programs
in the sample data file, so we may need some sort of cross-reference
check with institutions, though the naive approach to this would require
that source files be uploaded in a specific order, so we're going to
want to avoid that.

The Yellow Ribbon Program records corresponding with an Institution are
exposed as part of the institution profile endpoint. Swagger
documentation in Vets-API will be provided shortly to reference this new
functionality.

```
{
   ...
   "data" : {
      "id" : "1565",
      "attributes" : {
         "facility_code" : "21905928",
         ...
         "yellow_ribbon_programs" : [
            {
               "division_professional_school" : "All",
               "number_of_students" : 1000,
               "degree_level" : "All",
               "contribution_amount" : "10000.0"
            }
         ],
	...
}
```

Supersedes: #327 
Local Reference: #323 
Consumer: https://github.com/department-of-veterans-affairs/vets-website/pull/7654
Tracker Reference: https://github.com/department-of-veterans-affairs/vets.gov-team/issues/9365

cc @briangryth

### CODE REVIEW

File: db/migrate/20180603183002_create_yellow_ribbon_programs.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class CreateYellowRibbonPrograms < ActiveRecord::Migration
+  def change
+    create_table :yellow_ribbon_programs do |t|
+			t.integer :version, null: false, index: true
```
Comment (jkassemi) [inline db/migrate/20180603183002_create_yellow_ribbon_programs.rb]: @jkassemi - need to fix spacing here

File: app/models/institution.rb [hunk 2]
```diff
@@ -153,6 +153,8 @@ class Institution < ActiveRecord::Base
   validates :version, :institution, :country, presence: true
   validates :institution_type_name, inclusion: { in: TYPES }
 
+  has_many :yellow_ribbon_programs
```
Comment (saneshark) [inline app/models/institution.rb]: adding dependent destroy here makes sense I believe, unless this should really be a `has_many :through` relationship with a join table?

File: app/models/yellow_ribbon_program.rb [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# frozen_string_literal: true
+class YellowRibbonProgram < ActiveRecord::Base
+end
```
Comment (saneshark) [inline app/models/yellow_ribbon_program.rb]: should we define the `belongs_to` association?

File: spec/models/converters/date_converter_spec.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  it 'converts strings to dates' do
+    expect(subject.convert('4/17/2017')).to eq(Date.parse('April 17, 2017'))
+  end
+
+  it 'converts blank value to nil' do
+    expect(subject.convert('')).to be_nil
+  end
+
+  # TODO: we should find a way to pass warnings off for data that doesn't parse
+  it 'converts bad date to nil' do
```
Comment (saneshark) [inline spec/models/converters/date_converter_spec.rb]: is this TODO comment something that needs to be addressed prior to merge?

**Review by saneshark — COMMENTED** [review]
@jkassemi this looks good to me, but I had a few comments.

**Review by saneshark — APPROVED** [review]
This LGTM as is, but please see the outdated comments and changes I made. And of course lets wait on merging until tomorrow (post migration).

File: app/models/institution.rb (hunk 2 above)
Comment (jkassemi) [reply app/models/institution.rb]: 👍 should be good for the dependent destroy.

File: spec/models/converters/date_converter_spec.rb (hunk 4 above)
Comment (jkassemi) [reply spec/models/converters/date_converter_spec.rb]: I think we'll need to rethink the approach to the converters in order to surface these errors, so something that needs to be done later.

**Pull request merged**